src/java.base/share/classes/java/net/Socket.java

```diff
@@ -695,15 +708,9 @@ public void connect(SocketAddress endpoint, int timeout) throws IOException {
 
         try {
```

Comment:
At (old) L680, it checks that endpoint is an InetSocketAddress. We should also check that it resolved, as in:
```
        if (epoint.isUnresolved()) {
            close();
            throw new UnknownHostException(epoint.getHostName() + " is unresolved");
        }
```
The benefit is the the constructor does the validation before is delegates to the SocketImpl. We could change the SocketImpl to throw IAE for this case, it doesn't matter because it will never happen if Socket rejects it.

Reply:
Implemented in 424fc8ccd3a495a49f859f0cfc563cf463b3efeb, along with a test. Are you sure we want to have a `close()` there instead of `closeQuietly()`?

Reply:
I can't think of any cases where close could throw here but it would be better to catch any exception or error from close so that it doesn't override the UHE. A supporting method to catch and add as a suppressed exception is okay but maybe think more about the name as "quietly" it misleading, and "parentError" is a bit strange as it's not an Error.

Reply:
In a21c5e068555dc99e72d63ab29d3fbe75a543080,

* Renamed `closeQuietly()` to `closeSuppressingExceptions()`
* Replaced the `close()` usage with `closeSuppressingExceptions()` where we throw `UHE` in `connect()`